Microsoft.Research/Contracts/System/System.Diagnostics.FileVersionInfo.cs

```diff
@@ -0,0 +1,36 @@
+﻿
+
+// CodeContracts
+// 
+// Copyright (c) Microsoft Corporation
+// 
+// All rights reserved. 
+// 
+// MIT License
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System.Diagnostics.Contracts;
+using System;
+
+namespace System.Diagnostics
+{
+#if !SILVERLIGHT
```

Comment:
:question: What happens if we simply include this class when `SILVERLIGHT` is defined? If users aren't referencing this assembly directly, wouldn't everything still just work?

:bulb: When the whole type is excluded when `SILVERLIGHT` is defined, you can move the preprocessor directives to above the `using` declarations and at the end of the file, which will prevent someone from accidentally applying a Remove and Sort Usings operation that breaks the file in some context.